sdk/iot/core/inc/az_iot_hub_client.h

```diff
@@ -176,8 +176,8 @@ typedef struct az_iot_hub_client_properties
  * @brief Initializes the Telemetry or C2D properties.
  *
  * @param[in] properties The #az_iot_hub_client_properties to initialize
- * @param[in] buffer Can either be an empty #az_span or an #az_span containing properly formatted
- *                   properties starting with the '?' character.
+ * @param[in] buffer Can either be an empty #az_span or an #az_span containing properly formatted,
+ *                   RFC 2396-encoded properties with the following format: {key}={value}&{key}={value}.

```

Comment:
That RFC is obsolete. https://tools.ietf.org/html/rfc3986 instead. 

We need to understand exactly what _part_ of the RFC is required here.
Different parts of the URI require different types of encoding and usually most characters should not be percent-encoded.

The buffer must start with '?' either way - not sure why that part is removed.